src/lib/requestManager.test.ts

```diff
@@ -44,3 +95,134 @@ describe('requestManager', () => {
     requestSpy.mockRestore();
   });
 });
+
+describe('createRequestSuccessInterceptor', () => {
+  beforeEach(() => {
+    clearItem('UserProfile');
+  });
+
+  it('given an existing access_token, sets Authorization header in Axios config', async () => {
+    setItem('UserProfile', { auth: mockTokenData, user: mockUserProfileData });
+
+    const interceptor = createRequestSuccessInterceptor();
+    const config = interceptor({ headers: {}, url: endPoint });
+
+    expect(config.headers.Authorization).toBe(`Bearer ${mockTokenData.access_token}`);
+  });
+
+  it('given a non-existing access_token, does NOT set Authorization header in Axios config', async () => {
+    const interceptor = createRequestSuccessInterceptor();
+    const config = interceptor({ headers: {}, url: endPoint });
+
+    expect(config.headers.Authorization).toBeUndefined();
+  });
+});
+
+describe('createResponseErrorInterceptor', () => {
+  beforeEach(() => {
+    clearItem('UserProfile');
+
+    global.window = Object.create(window);
+
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: jest.fn(),
+      },
+      writable: true,
+    });
+  });
+
+  afterEach(() => {
+    window.location.href = windowHref;
+  });
+
+  it('given NON existing tokens and 401 error, redirects to login page', async () => {
+    const errorResponse = {
+      name: '',
+      message: '',
+      isAxiosError: true,
+      toJSON: () => ({}),
+      config: {},
+      code: '',
+      response: {
+        data: {},
+        status: 401,
+        statusText: '',
+        headers: {},
+        config: {},
+      },
+    };
+
+    try {
+      const interceptor = createResponseErrorInterceptor();
+      await interceptor(errorResponse);
+    } catch (err) {
+      expect(err).toBe(errorResponse);
+      expect(window.location.href).toEqual(LOGIN_URL);
+    }
+  });
+
+  // TODO:
+  // it('given existing refresh token and 401 error, updates access token with new', async () => {
```

Comment:
As discussed prefer to skip the tests instead of commenting out code

```suggestion
   xit('given existing refresh token and 401 error, updates access token with new', async () => {
```

Reply:
Ah it's `xit`, yeah I used `test.skip` when using `test()` on another but didn't think to look at the syntax when using `it` instead, thanks!

I left this for now as it seems the re-fetching of the token in the adapter fails, which is why the new token does not get set. I will revisit when knowledge is more.